File: app/src/main/java/com/nathaniel/motus/cavevin/view/CellarChoiceDialogFragment.java
package com.nathaniel.motus.cavevin.view;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Spinner;

import com.nathaniel.motus.cavevin.R;
import com.nathaniel.motus.cavevin.controller.MainActivity;
import com.nathaniel.motus.cavevin.model.Cellar;

/**
 * A simple {@link Fragment} subclass.
 */
public class CellarChoiceDialogFragment extends DialogFragment {

    private Spinner mCellarChoiceSpinner;
    private onCellarChoiceFragmentClickListener mCallBack;
    private String mDialogTitle;

    public CellarChoiceDialogFragment() {
        // Required empty public constructor
    }

    public interface onCellarChoiceFragmentClickListener {
        public void onCellarChoiceFragmentClick(int position);
    }

    public static CellarChoiceDialogFragment newInstance(String dialogTitle){

        CellarChoiceDialogFragment dialogFragment=new CellarChoiceDialogFragment();
        dialogFragment.mDialogTitle=dialogTitle;

        return dialogFragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        AlertDialog.Builder adb=new AlertDialog.Builder(getActivity());
        final LayoutInflater inflater=getActivity().getLayoutInflater();

        View view=inflater.inflate(R.layout.fragment_cellar_choice_dialog,null);
        mCellarChoiceSpinner =view.findViewById(R.id.fragment_cellar_choice_dialog_spinner);

        initializeCellarChoiceSpinner();

        adb.setView(view);
        adb.setTitle(mDialogTitle);
        adb.setPositiveButton("OK", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                int position= mCellarChoiceSpinner.getSelectedItemPosition();
                mCallBack.onCellarChoiceFragmentClick(position);
            }
        });
        adb.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                if (dialog!=null) dialog.dismiss();
            }
        });
        return adb.create();
    }

    private void initializeCellarChoiceSpinner(){

        ArrayAdapter<String> cellarNameListAdapter=new ArrayAdapter<String>(getActivity().getApplicationContext(),android.R.layout.simple_spinner_item);

        int i=0;
        while (i< Cellar.getCellarPool().size()){
            cellarNameListAdapter.add(Cellar.getCellarPool().get(i).getCellarName());
            i++;
        }

        cellarNameListAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        mCellarChoiceSpinner.setAdapter(cellarNameListAdapter);
        mCellarChoiceSpinner.setSelection(MainActivity.getCurrentCellarIndex());

        createCallBackToParentActivity();
    }

    private void createCallBackToParentActivity(){
        mCallBack=(onCellarChoiceFragmentClickListener)getActivity();
    }
}

File: app/src/main/java/com/nathaniel/motus/cavevin/view/CellarChoiceDialogFragment.kt
package com.nathaniel.motus.cavevin.view

import com.nathaniel.motus.cavevin.R
import android.os.Bundle
import com.nathaniel.motus.cavevin.model.Cellar
import com.nathaniel.motus.cavevin.controller.MainActivity
import android.app.AlertDialog
import android.app.Dialog
import com.nathaniel.motus.cavevin.R.layout
import android.widget.Spinner
import android.widget.ArrayAdapter
import androidx.fragment.app.DialogFragment

/**
 * A simple [Fragment] subclass.
 */
class CellarChoiceDialogFragment : DialogFragment() {
    private lateinit var mCellarChoiceSpinner: Spinner
    private var mCallBack: onCellarChoiceFragmentClickListener? = null
    private var mDialogTitle: String? = null

    interface onCellarChoiceFragmentClickListener {
        fun onCellarChoiceFragmentClick(position: Int)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val adb = AlertDialog.Builder(activity)
        val inflater = activity!!.layoutInflater
        val view = inflater.inflate(layout.fragment_cellar_choice_dialog, null)
        mCellarChoiceSpinner = view.findViewById(R.id.fragment_cellar_choice_dialog_spinner)
        initializeCellarChoiceSpinner()
        adb.setView(view)
        adb.setTitle(mDialogTitle)
        adb.setPositiveButton("OK") { dialog, which ->
            val position = mCellarChoiceSpinner.getSelectedItemPosition()
            mCallBack!!.onCellarChoiceFragmentClick(position)
        }
        adb.setNegativeButton("Cancel") { dialog, which -> dialog?.dismiss() }
        return adb.create()
    }

    private fun initializeCellarChoiceSpinner() {
        val cellarNameListAdapter = ArrayAdapter<String>(
            activity!!.applicationContext, android.R.layout.simple_spinner_item
        )
        var i = 0
        while (i < Cellar.cellarPool.size) {
            cellarNameListAdapter.add(Cellar.cellarPool[i].cellarName)
            i++
        }
        cellarNameListAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        mCellarChoiceSpinner!!.adapter = cellarNameListAdapter
        mCellarChoiceSpinner!!.setSelection(MainActivity.currentCellarIndex)
        createCallBackToParentActivity()
    }

    private fun createCallBackToParentActivity() {
        mCallBack = activity as onCellarChoiceFragmentClickListener?
    }

    companion object {
        fun newInstance(dialogTitle: String?): CellarChoiceDialogFragment {
            val dialogFragment = CellarChoiceDialogFragment()
            dialogFragment.mDialogTitle = dialogTitle
            return dialogFragment
        }
    }
}